pkg/obfuscate/sql.go

```diff
@@ -414,3 +422,52 @@ func (o *Obfuscator) ObfuscateSQLExecPlan(jsonPlan string, normalize bool) (stri
 	}
 	return o.sqlExecPlan.obfuscate([]byte(jsonPlan))
 }
+
+// ObfuscateWithSQLLexer obfuscates the given SQL query using the go-sqllexer package.
+// If ObfuscationMode is set to ObfuscateOnly, the query will be obfuscated without normalizing it.
+func (o *Obfuscator) ObfuscateWithSQLLexer(in string, opts *SQLConfig) (*ObfuscatedQuery, error) {
+	if opts.ObfuscationMode != ObfuscateOnly && opts.ObfuscationMode != ObfuscateAndNormalize {
+		return nil, fmt.Errorf("invalid obfuscation mode: %s", opts.ObfuscationMode)
+	}
+
+	obfuscator := sqllexer.NewObfuscator(
```

Comment:
Should we be building a whole new obfuscator for every call?

Reply:
Despite the name difference, the `NewObfuscator` works similar as the `NewTokenizer` in current implementation, which needs to be initialized with the input obfuscation options to tokenize and obfuscate.
https://github.com/DataDog/datadog-agent/blob/cce6815f47848a23edd72dd566628ae40abc82d4/pkg/obfuscate/sql.go#L320